tests/unit/OmiseTest.php

```diff
@@ -1,37 +1,121 @@
 <?php
-define('_PS_VERSION_', 'TEST_VERSION');
+if (! defined('_PS_VERSION_')) {
+    define('_PS_VERSION_', 'TEST_VERSION');
+}
 
 class OmiseTest extends PHPUnit_Framework_TestCase
 {
     private $omise;
+    private $setting;
+    private $smarty;
 
     public function setup()
     {
-        $paymentModule = $this->getMockBuilder(stdClass::class)
+        $this->getMockBuilder(stdClass::class)
             ->setMockClassName('PaymentModule')
-            ->setMethods(array('__construct', 'l'))
+            ->setMethods(
+                array(
+                    '__construct',
+                    'display',
+                    'displayConfirmation',
+                    'l',
+                )
+            )
+            ->getMock();
+
+        $this->setting = $this->getMockBuilder(Setting::class)
+            ->setMethods(
+                array(
+                    'getLivePublicKey',
+                    'getLiveSecretKey',
+                    'getSubmitAction',
+                    'getTestPublicKey',
+                    'getTestSecretKey',
+                    'getTitle',
+                    'isModuleEnabled',
+                    'isSandboxEnabled',
+                    'isSubmit',
+                    'save',
+                )
+            )
+            ->getMock();
+
+        $this->smarty = $this->getMockBuilder(stdClass::class)
+            ->setMockClassName('Smarty')
+            ->setMethods(
+                array(
+                    'assign',
+                )
+            )
             ->getMock();
 
         $this->omise = new Omise();
+        $this->omise->setSetting($this->setting);
+        $this->omise->setSmarty($this->smarty);
+    }
+
+    public function testConstructor_whenNewTheInstance_theDefaultValueOfTheAttributeSettingMustBeAvailable()
+    {
+        $omise = new Omise();
+
+        $setting = $omise->getSetting();
+
+        $this->assertNotEmpty($setting);
     }
 
-    public function testName_omise()
+    public function testName_theNameThatUsedToReferenceInTheProgramMustBe_omise()
     {
         $this->assertEquals('omise', $this->omise->name);
     }
 
-    public function testDisplayName_Omise()
+    public function testDisplayName_theNameThatUsedToDisplayeToTheMerchantMustBe_Omise()
     {
         $this->assertEquals('Omise', $this->omise->displayName);
     }
 
-    public function testNeedInstance_0()
+    public function testNeedInstance_noNeedToLoadModuleAtTheBackendModulePage()
     {
         $this->assertEquals(0, $this->omise->need_instance);
     }
 
-    public function testBootstrap_true()
+    public function testBootstrap_bootstrapTemplateIsRequired()
```

Comment:
btw, Im ok to skip that part if that "the expectation" is an optional, from
https://github.com/omise/omise-prestashop/pull/3#pullrequestreview-94529